### DISCUSSION TITLE
Fan Control takes a full minute to load with Corsair Commander Pro

### DISCUSSION THREAD

**Discussion by jonelsorel** [Rem0o/FanControl.Releases#1155] [Q&A] [answered]
Hello,

I have just bought a Corsair Commander Pro so I can set my pump at a fixed RPM and vary fans according to a graph with Fan Control, but I notice the software takes around 60 seconds to load the UI when launched manually or at startup (delay 1 sec). When removing the plugin, it loads almost instantly. I've tried unzipping the program from scratch to different folders/drives on my PC, but the story repeats itself. 

Dunno if anyone else is experiencing this. Here's hoping a fix would come soon, as it's a bit annoying to hear the fans spin like crazy for almost 2 minutes after bootup, until the software calms them down to the set levels. 

Thank you!

**Comment by Rem0o** [comment, marked as answer]
FanControl will refresh at startup up to 10 times until every sensor that is saved in the config shows up. Don't know about the specifics with the plugin, but I suspect the hid interface it uses takes a while to start showing the sensors, hence why FanControl is waiting.

**Reply by jonelsorel** [reply]
Thanks for the reply!  I'll try removing all the unnecessary sensor entries to see if that accelerates loading.

**Reply by Rem0o** [reply]
Not exactly. If the plugin takes time before it starts seeing the sensors, then FanControl will be blocked by it until it sees them back.  If your config depends on the plugin at all, you'll need to wait for the plugin to fully load.

**Reply by jonelsorel** [reply]
Got it, thanks again for the answer. I have a secondary dilemma after getting the Commander Pro. Fan Control fails to detect the unpopulated PWM channels no matter how many times I run assisted setup, and I'm not sure if this is normal software behavior or my fan hub is broken. I've tried "reinstalling" Fan Controls several times as well, doesn't change anything. Any tips would be greatly appreciated.
